### PR TITLE
Initial Vagrant setup

### PR DESCRIPTION
**Pull request by Hexcles**
I used the `Vagrantfile` in `taskcluster-client` as an example and changed the box version to 0.2.0, which is the latest I can find on our S3 (https://s3.amazonaws.com/task-cluster-dev/).

Yet it is still too old -- Ubuntu 13.10. I worked around the node/npm version problem by updating the local node/npm, but didn't really want to mess around with rabbitmq as it was installed from the repo. Tests are currently failing inside the Vagrant box because of outdated rabbitmq (the management API has changed).

Do we have a newer template box? Or shall I just go ahead and provision from the vanilla `hashicorp/precise64` (which wouldn't need much work)?

### CODE REVIEW

File: vagrant.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# Latest LTS as of Nov 22, 2016
+NODE_VERSION="6.9.1"
+
+# Install new nodejs.
+_node_filename="node-v${NODE_VERSION}-linux-x64.tar.xz"
+_node_url="https://nodejs.org/dist/v${NODE_VERSION}/${_node_filename}"
```
Comment (walac) [inline vagrant.sh]: How about [nvm](https://github.com/creationix/nvm)?

Comment (Hexcles) [inline vagrant.sh]: `nvm` is great, but we don't really benefit too much from it in this case as we don't need to switch the versions of nodejs.

Besides, nodejs is only part of the problem. The still unsolved problem is the outdated rabbitmq. If we have a newer base box, we can simply install node and rabbitmq from the repo and both problems should be solved automatically.

**Comment by imbstack** [comment]
@Hexcles I would just provision from `hashicorp/precise64`. I don't think there needs to be much tc specific stuff for this.

File: vagrant.sh (hunk 1 above)
Comment (Hexcles) [inline vagrant.sh]: I just tried `nvm` (because, unfortunately, even the nodejs in Ubuntu 16.04 is too old and has some features hidden behind the harmony flag), but found that it would actually compile node from source, which is quite slow. I guess I'll keep using the precompiled node.

Comment (Hexcles) [inline vagrant.sh]: Oops, never mind. I realize I probably made a mistake. `nvm` does use the precompiled binaries from the official website when possible. Sorry.

**Comment by Hexcles** [comment]
The new configuration uses the official `ubuntu/xenial64` box (the latest LTS). (I kept on forgetting those codenames.) @imbstack 

Xenial gives us a new rabbitmq, but node/npm in the official repo are still too old (some new features used in the project are behind the harmony flag). Thus, I use nvm to install the latest LTS node and enable it by default for user `ubuntu`, which is the default user upon `vagrant ssh`. @walac 

Tl;dr. Now you can
```
vagrant up
vagrant ssh
cd /vagrant
npm install && npm test
```

Cheers!
